internal/directory/github/github.go

```diff
@@ -230,6 +272,77 @@ func (p *Provider) getUser(ctx context.Context, userLogin string) (*apiUserObjec
 	return &res, nil
 }
 
+func (p *Provider) listUserOrganizationTeams(ctx context.Context, userSlug string, orgSlug string) ([]int, error) {
+	// GitHub's Rest API doesn't have an easy way of querying this data, so we use the GraphQL API.
```

Comment:
Wow this is wild. Nice work. I don't have a ton of experience with graphql but...uh, I kinda hope they don't go fully this direction.